cart/domain/decorator/cartDecorator.go

```diff
@@ -54,6 +56,20 @@ func (df *DecoratedCartFactory) Inject(
 	df.logger = logger
 }
 
+// Inject dependencies
+func (dci *DecoratedCartItem) Inject(
```

Comment:
Do we ever get Cart Items through DI?

Reply:
Good point! I removed the dependency injection and added the logger via factory methods